perceval/backends/core/rocketchat.py

```diff
@@ -0,0 +1,403 @@
+        of the channel, and returns a dict with the parsed data.
+
+        :param raw_channel_info: JSON string to parse
+
+        :returns: a dict with the parsed channel's information
+        """
+        result = json.loads(raw_channel_info)
+        return result['channel']
+
+    @classmethod
+    def has_archiving(cls):
+        """Returns whether it supports archiving items on the fetch process.
+
+        :returns: this backend supports items archive
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend supports items resuming
+        """
+        return True
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Rocket.Chat item."""
+
+        return item["_id"]
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a Rocket.Chat item.
+
+        The timestamp is extracted from 'ts' field,
+        and then converted into a UNIX timestamp.
+
+        :param item: item generated by the backend
+
+        :returns: extracted timestamp
+        """
+        ts = str_to_datetime(item['_updatedAt']).timestamp()
+        return ts
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a Rocket.Chat item.
+
+        This backend only generates one type of item which is
+        'message'.
+        """
+        return CATEGORY_MESSAGE
+
+    def _init_client(self, from_archive=False):
+        """Init client"""
+
+        return RocketChatClient(self.url, self.user_id, self.api_token,
+                                self.max_items, self.sleep_for_rate,
+                                self.min_rate_to_sleep, from_archive, self.archive, self.ssl_verify)
+
+
+class RocketChatClient(HttpClient, RateLimitHandler):
+    """Rocket.Chat API client.
+
+    Client for fetching information from the Rocket.Chat server
+    using its REST API.
+
+    :param url: server url from where messages are to be fetched
+    :param user_id: generated User Id using your Rocket.Chat account
+    :param api_token: token needed to use the API
+    :param max_items: maximum number of message requested on the same query
+    :param sleep_for_rate: sleep until rate limit is reset
+    :param min_rate_to_sleep: minimum rate needed to sleep until
+         it will be reset
+    :param from_archive: it tells whether to write/read the archive
+    :param archive: archive to store/retrieve items
+    :param ssl_verify: enable/disable SSL verification
+    """
+    RCHANNEL_MESSAGES = 'channels.messages'
+    RCHANNEL_INFO = 'channels.info'
+
+    HAUTH_TOKEN = 'X-Auth-Token'
+    HUSER_ID = 'X-User-Id'
+
+    PCHANNEL_NAME = 'roomName'
+    PCOUNT = "count"
+    POLDEST = "oldest"
+
+    def __init__(self, url=None, user_id=None, api_token=None, max_items=MAX_ITEMS,
```

Comment:
Similar comment as above, if url, user_id and api_token are mandatory, the default values are not needed.

Reply:
Thank you, I have updated it :)